tests/test_downloader/test_executors/test_base.py

```diff
@@ -0,0 +1,170 @@
+
+import re
+from unittest.mock import Mock, call
+
+from django.test import TestCase
+import pytest
+
+from downloader.models import DownloadRequest
+from downloader.executors.athena import AthenaExecutor
+from tests.factory import EntityFactory
+
+
+ef = EntityFactory()
+
+
+def normalize_query(q):
+    return re.sub(r'\s+', ' ', q).strip()
+
+
+class BaseExecutorTestCase(TestCase):
+
+    @pytest.fixture(autouse=True)
+    def initfixtures(self, mocker):
+        self.mocker = mocker
+
+        self.mocker.patch('downloader.executors.athena.sleep')
+
+    def setUp(self):
+        ef.clear()
+
+        self.executor = AthenaExecutor()
```

Comment:
athena? then why is it `BaseExecutorTestCase` ?

Reply:
unless you want to test BaseExecutor but you want to treat it as an abstract class then that's ok but then either create at the top of this file:
```python
class DummyExecutor(BaseExecutor):
    pass
``` 

and then test `DummyExecutor`, or since we're in python just create instances of `BaseExecutor`